pytest>=5.0.0 incompatible with Flask<=1.0.4

### Description

Master builds are failing due to error:
```
import_name = 'conftest'
    def find_package(import_name):
        """Finds a package and returns the prefix (or None if the package is
        not installed) as well as the folder that contains the package or
        module as a tuple.  The package path returned is the module that would
        have to be added to the pythonpath in order to make it possible to
        import the module.  The prefix is the path below which a UNIX like
        folder structure exists (lib, share etc.).
        """
        root_mod_name = import_name.split('.')[0]
        loader = pkgutil.get_loader(root_mod_name)
        if loader is None or import_name == '__main__':
            # import name is not found, or interactive/main module
            package_path = os.getcwd()
        else:
            # For .egg, zipimporter does not have get_filename until Python 2.7.
            if hasattr(loader, 'get_filename'):
                filename = loader.get_filename(root_mod_name)
            elif hasattr(loader, 'archive'):
                # zipimporter's loader.archive points to the .egg or .zip
                # archive filename is dropped in call to dirname below.
                filename = loader.archive
            else:
                # At least one loader is missing both get_filename and archive:
                # Google App Engine's HardenedModulesHook
                #
                # Fall back to imports.
                __import__(import_name)
                filename = sys.modules[import_name].__file__
            package_path = os.path.abspath(os.path.dirname(filename))
            # package_path ends with __init__.py for a package
>           if loader.is_package(root_mod_name):
E           AttributeError: 'AssertionRewritingHook' object has no attribute 'is_package'
.tox/py37-min/lib/python3.7/site-packages/flask/helpers.py:682: AttributeError
```

This can be tracked to following issues:
- https://github.com/pytest-dev/pytest/issues/5532
- https://github.com/pytest-dev/pytest/issues/5542

Indeed (either one of those actions):
- restricting `flask` to be at least `1.0.4` makes tests pass.
- restricting `pytest` to be at most `4.9.x` makes test pass

Which brings us down to final question.
Should `Flask` be kept at low version (which is pretty old TBH) or should `pytest` be restricted.
Personally I favor running newer version instead of supporting older codebases.

### Expected behaviour

master branch builds correctly against `min` environments.

### Actual behaviour

master branch builds correctly against `min` fail with aforementioned error.

### Steps to reproduce

Just run `tox -e py3(X)-min` where `X` stands for `Python` minor version {4,5,6,7}.

--- a/setup.py
+++ b/setup.py
@@ -33,7 +33,7 @@ install_requires = [
 ]
 
 swagger_ui_require = 'swagger-ui-bundle>=0.0.2'
-flask_require = 'flask>=0.10.1'
+flask_require = 'flask>=1.0.4'
 aiohttp_require = [
     'aiohttp>=2.3.10,<3.5.2',
     'aiohttp-jinja2>=0.14.0'

--- a/tests/decorators/test_security.py
+++ b/tests/decorators/test_security.py
@@ -64,14 +64,14 @@ def test_verify_oauth_scopes_remote(monkeypatch):
     session.get = get_tokeninfo_response
     monkeypatch.setattr('connexion.decorators.security.session', session)
 
-    with pytest.raises(OAuthScopeProblem, message="Provided token doesn't have the required scope"):
+    with pytest.raises(OAuthScopeProblem, match="Provided token doesn't have the required scope"):
         wrapped_func(request, ['admin'])
 
     tokeninfo["scope"] += " admin"
     assert wrapped_func(request, ['admin']) is not None
 
     tokeninfo["scope"] = ["foo", "bar"]
-    with pytest.raises(OAuthScopeProblem, message="Provided token doesn't have the required scope"):
+    with pytest.raises(OAuthScopeProblem, match="Provided token doesn't have the required scope"):
         wrapped_func(request, ['admin'])
 
     tokeninfo["scope"].append("admin")
@@ -102,14 +102,14 @@ def test_verify_oauth_scopes_local():
     request = MagicMock()
     request.headers = {"Authorization": "Bearer 123"}
 
-    with pytest.raises(OAuthScopeProblem, message="Provided token doesn't have the required scope"):
+    with pytest.raises(OAuthScopeProblem, match="Provided token doesn't have the required scope"):
         wrapped_func(request, ['admin'])
 
     tokeninfo["scope"] += " admin"
     assert wrapped_func(request, ['admin']) is not None
 
     tokeninfo["scope"] = ["foo", "bar"]
-    with pytest.raises(OAuthScopeProblem, message="Provided token doesn't have the required scope"):
+    with pytest.raises(OAuthScopeProblem, match="Provided token doesn't have the required scope"):
         wrapped_func(request, ['admin'])
 
     tokeninfo["scope"].append("admin")

